File: sorting_algorithm/selection_sort.dart


int findSmallest(List<int> lists){

  int smallest = lists[0];
  /* int smallestIndex = 0; */

  for(int i = 1; i < lists.length; i++){

    if(lists[i] < smallest){
      smallest = lists[i];
      /* smallestIndex = i; */
    }

  }

  return smallest;

}


List<int> selectionSort(List<int> lists){

  List<int> newList = [];


  for(int i = 0; i < lists.length; i++){

    int smallest = findSmallest(lists);

    newList.add(smallest);

    lists.remove(smallest);


  }


  return newList;


}


void main(){

  print(selectionSort([7,4,19,33,2,10,48,22,13,1,88,0,92,30,6]));

  /* print(findSmallest([6,3,88,90,1,17,60,40,30,20,15,99,1000])); */

}

File: sorting_algorithm/selection_sort.py
def search_smallest(lists: list[int]) -> int:
    smallest: int = lists[0]
    
    for i in range(len(lists)):
        if lists[i] < smallest:
            smallest = lists[i]
            
    
    return smallest

    




def selection_sort(lists: list[int]) -> list[int]:
    
    new_list: list[int] = []
    
    for i in range(len(lists)):
        smallest: int = search_smallest(lists)
        new_list.append(lists.pop(smallest))
    
    return new_list
    
    

    
    
    
print(selection_sort([5,28,12,90,1,33,4,16,9,32,10]))

    